### DISCUSSION TITLE
Beta [Pre-release] Release Status Clarification

### DISCUSSION THREAD

**Discussion by BipedalPrimate** [Louisvdw/dbus-serialbattery#994] [Q&A] [answered]
Outstanding effort in the development & updating of this software - thanks to all involved.

Having been a user for 3 months or so, I am now ready to consider when to upgrade from the v1.0.20230531 [Latest].  Indeed, some of the features included since then will be of benefit.

Since that release, there have been 4 Beta [Pre-release] versions.

In a past life in the commercial world, I had always tried to stay away from Beta [Pre-release] versions in a production environment as my job was on the line.  These days, it is my quality of life that is on the line - thinking back on it, I used to think the job was more important :-)

GitHub describes Pre-release as "not ready for production and may be unstable". [https://docs.github.com/en/repositories/releasing-projects-on-github/managing-releases-in-a-repository]

The most recent Pre-release (v1.2.20240227beta) includes a note that "The Bluetooth and CAN connections are still not stable on some systems. If you want to have a stable connection use the serial connection."

Should I conclude that the note describes the only _**known**_ instability concerns?  

Conversely, if I am not using Bluetooth or CAN, should I consider the most recent release to be [Latest] and stable enough for a production environment?

Being new to Open Source & GitHub, I am seeking clarity on how to assess the releases.

**Comment by mr-manuel** [comment, marked as answer]
> GitHub describes Pre-release as "not ready for production and may be unstable". [https://docs.github.com/en/repositories/releasing-projects-on-github/managing-releases-in-a-repository]

For sure not every repo is following this at 100%. All versions and changes I make are tested and should never leave the system completely unstable. Maybe it's not charging at full power or not going to float but else there should not be big issues.

> Should I conclude that the note describes the only known instability concerns?

Yes. Since there are also lots of different systems and configurations out there I added this note. For some Bluetooth works very reliable and for some absolutely not. But the serial connection work good for all and is the main focus.

Check also this page to get a bit more informations about the different releases: https://louisvdw.github.io/dbus-serialbattery/general/install#latest-release
